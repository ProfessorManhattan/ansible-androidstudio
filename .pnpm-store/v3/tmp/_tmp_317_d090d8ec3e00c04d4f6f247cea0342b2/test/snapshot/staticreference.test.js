const parser = require("../main");

describe("staticreference", function () {
  it("variable", function () {
    expect(parser.parseEval("static::$var;")).toMatchSnapshot();
  });
  it("constant", function () {
    expect(parser.parseEval("static::CONSTANT;")).toMatchSnapshot();
  });
  it("call", function () {
    expect(parser.parseEval("static::call();")).toMatchSnapshot();
  });
  it("uppercase", function () {
    expect(parser.parseEval("STATIC::call();")).toMatchSnapshot();
  });
});
