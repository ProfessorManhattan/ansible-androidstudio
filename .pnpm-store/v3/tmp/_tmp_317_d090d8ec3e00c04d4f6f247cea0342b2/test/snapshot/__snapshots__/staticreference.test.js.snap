// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`staticreference call 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [],
        "kind": "call",
        "what": StaticLookup {
          "kind": "staticlookup",
          "offset": Identifier {
            "kind": "identifier",
            "name": "call",
          },
          "what": StaticReference {
            "kind": "staticreference",
            "raw": "static",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`staticreference constant 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": StaticLookup {
        "kind": "staticlookup",
        "offset": Identifier {
          "kind": "identifier",
          "name": "CONSTANT",
        },
        "what": StaticReference {
          "kind": "staticreference",
          "raw": "static",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`staticreference uppercase 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [],
        "kind": "call",
        "what": StaticLookup {
          "kind": "staticlookup",
          "offset": Identifier {
            "kind": "identifier",
            "name": "call",
          },
          "what": StaticReference {
            "kind": "staticreference",
            "raw": "STATIC",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`staticreference variable 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": StaticLookup {
        "kind": "staticlookup",
        "offset": Variable {
          "curly": false,
          "kind": "variable",
          "name": "var",
        },
        "what": StaticReference {
          "kind": "staticreference",
          "raw": "static",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
